File: src/main/java/com/trade/securities/application/service/FscNewShareCertificateService.java
package com.trade.securities.application.service;

import com.trade.securities.application.port.in.GetFscNewShareCertificateUseCase;
import com.trade.securities.application.port.out.LoadFscNewShareCertificatePort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Map;

@Service
@RequiredArgsConstructor
public class FscNewShareCertificateService implements GetFscNewShareCertificateUseCase {

    private final LoadFscNewShareCertificatePort loadFscNewShareCertificatePort;

    @Override
    public Map<String, Object> getNewShareCertificates(int pageNo, int numOfRows, String basDt, String itmsNm,
            String likeItmsNm) {
        return loadFscNewShareCertificatePort.loadNewShareCertificates(pageNo, numOfRows, basDt, itmsNm, likeItmsNm);
    }
}

File: src/main/kotlin/com/trade/securities/application/service/FscNewShareCertificateService.kt
package com.trade.securities.application.service

import com.trade.securities.application.port.`in`.GetFscNewShareCertificateUseCase
import com.trade.securities.application.port.out.LoadFscNewShareCertificatePort
import org.springframework.stereotype.Service

@Service
class FscNewShareCertificateService(
    private val loadFscNewShareCertificatePort: LoadFscNewShareCertificatePort
) : GetFscNewShareCertificateUseCase {

    override fun getNewShareCertificates(pageNo: Int, numOfRows: Int, basDt: String, itmsNm: String?, likeItmsNm: String?): Map<String, Any?> {
        return loadFscNewShareCertificatePort.loadNewShareCertificates(pageNo, numOfRows, basDt, itmsNm, likeItmsNm)
    }
}
